packages/theme-default/src/layout/DocLayout/index.tsx

```diff
@@ -74,7 +74,7 @@ export function DocLayout(props: DocLayoutProps) {
       <div
         className={`${styles.content} rspress-doc-container flex flex-shrink-0 mx-auto`}
       >
-        <div className="w-full">
+        <div className="w-full overflow-hidden flex-1">
```

Comment:
@sumy7 can we remove the `overflow-hidden` or add a new class-name to this element?

The Rsbuild document is broken by the `overflow-hidden` style, as Rsbuild uses a custom overview component in this page: https://deploy-preview-1713--rsbuild.netlify.app/config/index

<img width="1350" alt="Screenshot 2024-03-03 at 21 51 30" src="https://github.com/web-infra-dev/rspress/assets/7237365/8deb2ec6-bb94-4492-b723-a09af27951ce">

Reply:
@chenjiahan removing overflow-hidden sounds like a good idea.
@sanyuan0704 , could you help fix this ? Or I will  submit a new PR tomorrow.

Also, for this page, I suggest disabling the display of the outline in the frontmatter to achieve a full-screen effect.

Reply:
The `overflow-hidden` style seems redundant. Can you explain the influence after removing it? Perhaps there are some cases i didn't notice. And i'm not convenient now, expect your new PR! ❤️

Reply:
Hi @sanyuan0704.
Let me explain the purpose of `overflow-hidden` here.
When I initially added `overflow-hidden`, the intention was to use the BFC to prevent long, non-wrapping sentences in child elements from exceeding the restricted width and compressing the outline.
However, following @chenjiahan 's suggestion, I switched to using a fixed width and added `flex-1` to ensure the space occupied by the outline.
As a result, the original purpose of `overflow-hidden` to limit the width of long sentences was lost.

This style layout modification has brought up many cases that I hadn't considered, as reported by @chenjiahan from rsbuild. However, the approach of fixing problems as they arise can easily shake my confidence. I'm considering how to avoid situations where fixing one issue leads to more problems.

Reply:
Thanks for your detailed explanation!

As you said, it is very easy to introduce some small breaking changes or edge cases when modifying CSS styles. It is also hard to testing it.

From my perspective, we can do :

1. Stabilize the main CSS styles and avoiding changing them frequently.
2. Add more class names to main elements to allow users overriding the styles.